Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -1,26 +1,62 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Runtime.ExceptionServices;
+using System;
+using System.Reflection.Metadata.Ecma335;
+
+namespace BootCamp.Chapter
 {
     public class Inventory
     {
-        private Item[] _items;
-        public Item[] GetItems()
+        protected Item[] _items;
+
+        // displays all items in inventory
+        // renamed from 'GetItems' to show the clear different from the 'GetItems' method below
+        public void ShowAllItems()
         {
-            return new Item[0];
+            if (!Array.TrueForAll(_items, IsNull => IsNull == null))
+            {
+                Console.WriteLine("Items in inventory:\n");
+                foreach (var item in _items)
+                {
+                    if (item != null)
+                    {
+                        Console.WriteLine(item.GetName());
+                    }
+                    else
+                    {
+                        continue;
+                    }
+                }
+            }
+            else
+            {
+                Console.WriteLine("The inventory is currently empty.");
+            }
         }
 
-        public Inventory()
+        public Item GetItem(string name)
         {
-            _items = new Item[0];
+            return _items[Array.FindIndex(_items, element => element.GetName().Equals(name))];         
```

Comment:
It does work, but in general you should treat C# as raw, without all the neat language features.